db/warehouse/migrate/20230301172341_add_custom_assessments.rb

```diff
@@ -0,0 +1,82 @@
+class AddCustomAssessments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :CustomAssessments do |t|
+      t.string :EnrollmentID, null: false
+      t.string :PersonalID, null: false
+      t.string :UserID, limit: 32, null: false
+      t.date :AssessmentDate, null: false
+      t.integer :DataCollectionStage, null: false, comment: 'One of the HMIS 5.03.1, or 99 for local use'
+      t.integer :data_source_id
+
+      t.datetime :DateCreated, null: false
+      t.datetime :DateUpdated, null: false
+      t.datetime :DateDeleted
+    end
+
+    create_table :CustomClientAssessments do |t|
+      t.string :PersonalID, null: false
+      t.string :UserID, limit: 32, null: false
+      t.date :InformationDate, null: false
+      t.integer :data_source_id
+
+      t.datetime :DateCreated, null: false
+      t.datetime :DateUpdated, null: false
+      t.datetime :DateDeleted
+    end
+
+    create_table :CustomProjectAssessments do |t|
+      t.string :ProjectID, null: false
+      t.string :UserID, limit: 32, null: false
+      t.date :InformationDate, null: false
+      t.integer :data_source_id
+
+      t.datetime :DateCreated, null: false
+      t.datetime :DateUpdated, null: false
+      t.datetime :DateDeleted
+    end
+
+    rename_table :hmis_assessment_processors, :hmis_form_processors
+
+    # Commented-out: if hmis_assessment_processors doesn't exist, run this instead
+    # create_table :hmis_form_processors do |t|
+    #   t.references :enrollment_coc
+    #   t.references :health_and_dv
+    #   t.references :income_benefit
+    #   t.references :physical_disability
+    #   t.references :developmental_disability
+    #   t.references :chronic_health_condition
+    #   t.references :hiv_aids
+    #   t.references :mental_health_disorder
+    #   t.references :substance_use_disorder
+    #   t.references :exit
+    # end
```

Comment:
```suggestion
    GrdaWarehouseBase.connection.table_exists? 'hmis_assessment_processors'
      rename_table :hmis_assessment_processors, :hmis_form_processors
    else
      create_table :hmis_form_processors do |t|
        t.references :enrollment_coc
        t.references :health_and_dv
        t.references :income_benefit
        t.references :physical_disability
        t.references :developmental_disability
        t.references :chronic_health_condition
        t.references :hiv_aids
        t.references :mental_health_disorder
        t.references :substance_use_disorder
        t.references :exit
      end
    end
```

Reply:
Ah cool. Thanks!

Reply:
This may need a bit more adjustment to work correctly in both up/down, and maybe a check to see if `hmis_form_processors` already exists....

Reply:
Alright, I can test it out. I think I need to add separate `up`/`down` blocks to make this reversible, I didn't think rails was automatically dropping the tables when it is brought down. We don't need to preserve any data, since all the data in the processor table was dropped in the previous migration.

Reply:
Alright, I pushed a change to make it reversible. I don't think we should expect any weirdness with the processor table in particular, I just had that comment there because I was messing with running and re running the migration a bunch locally.